src/fromager/commands/lint_requirements.py

```diff
@@ -27,10 +28,16 @@ def lint_requirements(input_files_path: list[click.Path]) -> None:
     flag = True
 
     for path in input_files_path:
-        parsed_lines = requirements_file.parse_requirements_file(str(path))
+        parsed_lines = requirements_file.parse_requirements_file(path)
+        unique_entries: set[str] = set()
         for line in parsed_lines:
             try:
-                Requirement(line)
+                requirement = Requirement(line)
+                if requirement.name in unique_entries:
+                    raise InvalidRequirement("Duplicate entry")
+                unique_entries.add(requirement.name)
+                if requirement.extras and path.name.endswith("constraints.txt"):
```

Comment:
We should document the way this naming convention is handled in the command help.

Reply:
Here is the text @lbrabec add added to the command help
```
It checks the formatting of these files and reports issues if found. Files with names that
    end with constraints.txt (e.g. constraints.txt, global-constraints.txt, etc.) are not allowed
    to contain extra dependencies.
```

Reply:
@lbrabec It is good practice to add comments or respond to the original comment so that it is clear how this is fixed before marking it resolved.